pkg/api/convertertoapi.go

```diff
@@ -611,6 +615,21 @@ func convertVLabsOrchestratorProfile(vp *vlabs.Properties, api *OrchestratorProf
 	}
 }
 
+func rationalizeReleaseAndVersionForDeployExceptForInvalidOnes(orchestratorProfile *vlabs.OrchestratorProfile, isUpdate bool, hasWindows bool) string {
+	version := common.RationalizeReleaseAndVersion(
+		orchestratorProfile.OrchestratorType,
+		orchestratorProfile.OrchestratorRelease,
+		orchestratorProfile.OrchestratorVersion,
+		isUpdate,
+		hasWindows)
+	//if we didn find one, it's invalid, but we don want to get the default one, so reset to the provided version
+	//validation kicks in later for deploy and will fail.
+	if version == "" {
+		return orchestratorProfile.OrchestratorVersion
```

Comment:
I don't think we want to simply return orchestratorProfile.OrchestratorVersion.

Here's an example where that wouldn't work:
If I set my `OrchestratorRelease` to "1.10" and my `OrchestratorVersion` to "1.11.6" -> `RationalizeReleaseAndVersion` won't be able to rationalize the release with the version since they are contradictory so it will return an empty version string. But "1.11.6"  in itself is a valid version so that won't fail validation later. Unless we call RationalizeReleaseAndVersion later on but in that case there is no need for this duplicate check all-together.

Reply:
addressed